evergreen/eg_app/consumers.py

```diff
@@ -0,0 +1,238 @@
+                    await self.channel_layer.group_send(
+                        self.room_group_name,
+                        {
+                            "type": "like_update",
+                            "post_id": post_id,
+                            "likes": post_data["likes"],
+                            "likers_display": post_data["likers_display"],
+                            "has_liked": post_data["has_liked"],
+                        },
+                    )
+            elif data["type"] == "upload_post":
+
+                post = await self.handle_post_upload(data)
+
+                if post:
+                    # BROADCAST TO ALL CLIENTS
+                    posts_data = await self.get_all_posts()
+
+                    await self.channel_layer.group_send(
+                        self.room_group_name,
+                        {
+                            "type": "feed_update",
+                            "data": {"posts": posts_data},
+                        },
+                    )
+
+                    # send upload
+                    await self.send(
+                        text_data=json.dumps(
+                            {
+                                "type": "upload_response",
+                                "status": "success",
+                                "post_id": str(post.id),
+                            }
+                        )
+                    )
+        except Exception as e:
+            print(f"Error in receive: {str(e)}")
+
+    @database_sync_to_async
+    def get_all_posts(self):
+
+        Post = apps.get_model("eg_app", "Post")
+
+        posts = Post.objects.all().order_by("-timestamp")
+
+        all_posts_data = []
+
+        for post in posts:
+            current_user = self.scope["user"]
+
+            post_image_url = ""
+
+            if post.image and hasattr(post.image, "url"):
+                post_image_url = post.image.url
+
+            user_has_liked = False
+
+            if current_user.is_authenticated:
+                user_has_liked = post.userLikes.filter(id=current_user.id).exists()
+
+            post_data = {
+                "id": str(post.id),
+                "user": post.user,
+                "image": {"url": post_image_url},
+                "caption": post.caption + "\n",  # Add newline after caption
+                "likes": post.likes,
+                "likers_display": post.get_likers_display(),
+                "has_liked": user_has_liked,
+                "comments": [],  # start w/ empty!
+            }
+
+            all_posts_data.append(post_data)
+
+        # Return the complete list of post data
+        return all_posts_data
+
+    @database_sync_to_async
+    def handle_post_upload(self, data):
+        Post = apps.get_model("eg_app", "Post")
+
+        try:
+            user = self.scope["user"]
+            if str(user.username) == "AnonymousUser":
+                raise Exception("Must be logged in to post")
```

Comment:
This does not prevent `Error creating post: 'AnonymousUser' object has no attribute 'email'`, and raising an exception like this still would not send feedback to the user